0024-pre-shared-key-v2/0024-pre-shared-key-v2.md

```diff
@@ -0,0 +1,266 @@
+---
+title: Pre-Shared Key V2 (PSKv2) Transport Protocol
+draft: 1
+---
+# Pre-Shared Key V2 (PSKv2) Transport Protocol
+
+This spec defines version 2 of the Pre-Shared Key Transport Protocol. PSKv2 implements end-to-end data encryption, condition and fulfillment generation, [End-to-End Quoting](#end-to-end-quoting) and [Chunked Payments](#chunked-payments). The approach to chunked payments is inspired by the Internet's Transmission Control Protocol (TCP).
+
+## Overview
+
+Like [PSKv1](./0016-pre-shared-key/0016-pre-shared-key.md), PSKv2 uses a secret shared between the sender and receiver to generate the conditions and fulfillments for payments without repeated end-to-end communication. PSKv2 additionally includes support for End-to-End Quoting and Chunked Payments.
+
+This specificiation aims to standardize the behavior of receivers while allowing for sophisticated sender behavior to be developed in the future. [Appendix A: Advanced Sender Features](#appendix-a-advanced-sender-features) outlines areas for potential sender features to be implemented in the future.
+
+## Setup
+
+Before using PSKv2, the sender and receiver use an authenticated, encrypted communication channel (generally provided as part of an [Application Layer Protocol](./0001-interledger-architecture/0001-interledger-architecture.md#application-layer)) to exchange:
+
+- PSK version (such as `2.0` for this version)
+- A 32-byte random or pseudorandom shared secret
+- The receiver's ILP address
+
+The receiver MAY generate the shared secret from a longer-term secret and a nonce that they put into the ILP address they give to the sender, as described in [PSKv1's Appendix A: Recommended Algorithm for Generating Shared Secrets](./0016-pre-shared-key/0016-pre-shared-key.md#appendix-a-recommended-algorithm-for-generating-shared-secrets).
```

Comment:
> [...] longer-term secret and a nonce that they put into the ILP address [...]

Consider rewording to make it clear that only the nonce goes into the ILP address, not the long-term secret.